js/test/wallet/createTransaction.test.ts

```diff
@@ -0,0 +1,194 @@
+                expect.assertions(1);
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'),
+                        new BigNumber(1.0000000000001),
+                        {}
+                    );
+                }).rejects.toThrow('amount decimals invalid');
+            });
+
+            it('throws an error on insufficient funds', async () => {
+                expect.assertions(1);
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'),
+                        new BigNumber(100),
+                        {}
+                    );
+                }).rejects.toThrow('insufficient funds');
+            });
+        });
+    });
+
+    describe('when transferring SPL token', () => {
+        describe('errors', () => {
+            it.todo('throws an error on invalid mint');
+
+            it('throws an error on invalid decimal amount', async () => {
+                expect.assertions(1);
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'),
+                        new BigNumber(1.0000000001),
+                        {
+                            token: new PublicKey('8FRFC6MoGGkMFQwngccyu69VnYbzykGeez7ignHVAFSN'),
+                        }
+                    );
+                }).rejects.toThrow('amount decimals invalid');
+            });
+
+            // FIXME: can't get past this error
+            // Fails at getting associated token account
+            it.skip('throws an error on unintialized payer', async () => {
+                expect.assertions(1);
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'),
+                        new BigNumber(100),
+                        {
+                            token: new PublicKey('8FRFC6MoGGkMFQwngccyu69VnYbzykGeez7ignHVAFSN'),
+                        }
+                    );
+                }).rejects.toThrow('payer not initialized');
+            });
+
+            it.todo('throws an error on frozen payer');
+
+            it.todo('throws an error on unitialized recipient');
+
+            it.todo('throws an error on frozen recipient');
+
+            it.todo('throws an error on insufficient funds');
```

Comment:
Actually, I wonder if we should try to mock the return values of `getAccount` etc. for these and for the checks above.

Reply:
Yup, that's what I was leaning toward for these ones

Reply:
I'm thinking we should maybe do this for anything that needs the SPL token functions. It's better if the tests don't require running a local validator / devnet

Reply:
I think we should decide on a single approach either way -- mock everything or nothing.

Argument for: things like `connection.getTransaction` in `validateTransactionSignature` might be easier to mock than to create the necessary state on chain. Tests will be faster and don't require running a local validator.

Argument against: if we mock it and there are breaking changes, we don't discover them, and it's also annoying to mock things.

Reply:
yup, makes sense

Reply:
I think for these unit tests, we mock them. We can have one integration test that will catch these breaking changes

Reply:
Sounds good. Let's mock everything then! We can worry about integration tests after